Mods/Alpakit/Source/Alpakit/Private/AlpakitReleaseWidget.cpp

```diff
@@ -9,298 +9,348 @@
+								{
+									return FText::Format(LOCTEXT("UpdateSMLVersionTooltip", "This mod depends on SML {0}, but the project is using SML {1}. Click to update"), FText::FromString(GetModSMLDependencyVersion(Mod)), FText::FromString(TargetSMLVersion));
+								})
+								.OnClicked_Lambda([this, Mod, TargetSMLVersion]
+								{
+									SetModSMLDependencyVersion(Mod, TargetSMLVersion);
+									return FReply::Handled();
+								})
+							]
+						]
+						+ SHorizontalBox::Slot().AutoWidth().Padding(5,0)[
+							SNew(SBox)
+							.HAlign(HAlign_Center)
+							.Visibility_Lambda([this, Mod]
+							{
+								FVersion GameVersion;
+								FVersionRange ModGameVersionRange, TargetGameVersionRange;
+								GetModGameVersionFields(Mod, GameVersion, ModGameVersionRange, TargetGameVersionRange);
+								if (FormatGameVersionRange(ModGameVersionRange) == FormatGameVersionRange(TargetGameVersionRange))
+									return EVisibility::Hidden;
+								return EVisibility::Visible;
+							})
+							.Content()[
+								SNew(SButton)
+								.Content()
+								[
+									SNew(SImage)
+									.Image(FAlpakitStyle::Get().GetBrush("Alpakit.Warning"))
+								]
+								.ToolTipText_Lambda([Mod]
+								{
+									FString GameVersionRaw = GetModGameVersion(Mod);
+									FVersion GameVersion;
+									FVersionRange ModGameVersionRange, TargetGameVersionRange;
+									GetModGameVersionFields(Mod, GameVersion, ModGameVersionRange, TargetGameVersionRange);
+									if (GameVersionRaw.IsEmpty()) {
+										GameVersionRaw = "(unspecified)";
+									} else {
+										GameVersionRaw = FormatGameVersionRange(ModGameVersionRange);
+									}
+									return FText::Format(LOCTEXT("UpdateGameVersionTooltip", "This mod uses game version {0}, but the project is {1}. Click to update"), FText::FromString(GameVersionRaw), FText::FromString(FormatGameVersionRange(TargetGameVersionRange)));
+								})
+								.OnClicked_Lambda([this, Mod]
+								{
+									FVersion GameVersion;
+									FVersionRange ModGameVersionRange, TargetGameVersionRange;
+									GetModGameVersionFields(Mod, GameVersion, ModGameVersionRange, TargetGameVersionRange);
+									SetModGameVersion(Mod, FormatGameVersionRange(TargetGameVersionRange));
+									return FReply::Handled();
+								})
+							]
+						]
+						//Box here for button to open output folder, if exists (ie packaged)
+						+ SHorizontalBox::Slot().AutoWidth().Padding(5, 0)[
+							SNew(SBox)
+							.HAlign(HAlign_Right)
+							.Visibility_Lambda([this, Mod]
+								{
+								   FString archivePath = FPaths::ConvertRelativePathToFull(FPaths::Combine(FPaths::ProjectSavedDir(), TEXT("ArchivedPlugins"), Mod->GetName()));
+									if (FPaths::DirectoryExists(archivePath))
+										return EVisibility::Visible;
+									return EVisibility::Hidden;
+								})
+							.Content()[
+								SNew(SButton)
+								.Text(LOCTEXT("OpenDirAlphakit", "Open Folder"))
```

Comment:
Welp, didn't know committing just the suggestion would immediately end the conversation!

> `FPlatformProcess::ExploreFolder` seems to support highlighting a file in the file explorer (by just passing a file path instead of a directory), so I think highlighting the release (`{ModName}.zip`) file specifically would help reduce the confusion about the multiple zips in that folder too.

I'll try that! If it works (I hope), it'll definitely be used.
> There's quite a bit of code duplication with the path computation, the ArchivedPlugins full path could be computed once in the Construct function and passed on to the lambdas that use it. The mod archive path can also be computed once in `ModEntryTrail_Lambda` (using the already computed ArchivedPlugins path) and passed on.

> I was a bit worried about the visibility function, because that's called every tick, so that's a lot of file existence checks. A quick profile (8 seconds) showed 1.1ms spent in `FPaths::DirectoryExists`, but then 172ms in `FWindowsPlatformFile::DirectoryExists`, which was only called from `FPaths::DirectoryExists`, so not sure which of those to trust, but a test with 700 mod entries didn't visually show much of a difference in performance, the editor was already slow just with the long list.

Doing the root computation in Construct to de-duplicate code wasn't even noticed, thanks!
From what I've gathered about the FPaths' functions, many are shuffled off to the proper Platform's native functions(FPaths, etc. are front-ends), that's why most of the time is spent in `FWindowsPlatformFile::DirectoryExists`.

Would it be better to check for the existence of `{ModName}.zip` instead of just the directory? From what I can tell now, the dir may exist without the packaged mod in the case the build/zipping failed.